src/main/java/io/vertx/gsoc2023/qotd/QoTDHandlers.java

```diff
@@ -0,0 +1,37 @@
+package io.vertx.gsoc2023.qotd;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.pgclient.PgPool;
+
+public record QoTDHandlers(PgPool pool) {
+
+  public void getAllQuotes(RoutingContext ctx) {
+    fetchAllQuotes()
+      .onFailure(Throwable::printStackTrace)
```

Comment:
Can you propose something else to convey the error to the user?

Besides, with this code, in case of error, the client will wait indefinitely for a reply.

Reply:
Just shipped it. I think it is much better now.